src/state/MediaDevices.ts

```diff
@@ -93,15 +94,16 @@ export const iosDeviceMenu$ = navigator.userAgent.includes("iPhone")
 
 function availableRawDevices$(
   kind: MediaDeviceKind,
-  usingNames$: Observable<boolean>,
+  updateAvailableDeviceRequests$: Observable<boolean>,
   scope: ObservableScope,
 ): Observable<MediaDeviceInfo[]> {
-  return usingNames$.pipe(
-    switchMap((usingNames) =>
+  return updateAvailableDeviceRequests$.pipe(
+    startWith(false),
+    switchMap((withPermissions) =>
```

Comment:
Instead of having updateAvailableDeviceRequests$, we could stick with usingNames$ and just expand the logic here to do some "trial and error" by creating a device observer without permissions and then re-creating it if we need names which aren't present.

Or, since this is getting kind of hairy, maybe we should just submit a fix upstream to LiveKit Components :thinking: